module/VuFindSearch/src/VuFindSearch/Backend/LibGuides/Connector.php

```diff
@@ -187,15 +196,35 @@ protected function process($data)
 
         $items = [];
 
-        // Extract titles and URLs from response:
-        $regex = '/<a href="([^"]*)"[^>]*>([^<]*)</';
-        $count = preg_match_all($regex, $data, $matches);
+        $itemRegex = '/<li>(.*?)<\/li>/';
+        $linkRegex = '/<a href="([^"]*)"[^>]*>([^<]*)</';
+        $descriptionRegex = '/<div class="s-lg-(?:widget|guide)-list-description">(.*?)<\/div>/';
+
+        // Extract each result item
+        $itemCount = preg_match_all($itemRegex, $data, $itemMatches);
 
-        for ($i = 0; $i < $count; $i++) {
-            $items[] = [
-                'id' => $matches[1][$i],    // ID = URL
-                'title' => $matches[2][$i],
+        for ($i = 0; $i < $itemCount; $i++) {
+            // Extract the link, which contains both the title and URL.
+            $linkCount = preg_match_all($linkRegex, $itemMatches[1][$i], $linkMatches);
+            if ($linkCount != 1) {
+                throw new \Exception('LibGuides result item included more than one link: ' . $itemMatches[1][$i]);
+            }
+            $item = [
+                'id' => $linkMatches[1][0],    // ID = URL
+                'title' => $linkMatches[2][0],
             ];
+
+            // Extract the descrpition.
+            if ($this->displayDescription) {
+                $descriptionCount = preg_match_all($descriptionRegex, $itemMatches[1][$i], $descriptionMatches);
+                if ($descriptionCount > 1) {
+                    throw new \Exception('LibGuides result item included more than one description: ' .
```

Comment:
Does this need to be a fatal exception? Might it be better to log a warning and just proceed without a description?